corehq/motech/repeaters/models.py

```diff
@@ -251,8 +314,398 @@ def reset_next_attempt(self):
             self.next_attempt_at = None
             self.save()
 
+    def register(self, payload, fire_synchronously=False):
+        if not self.allowed_to_forward(payload):
+            return
+
+        now = datetime.utcnow()
+        repeat_record = RepeatRecord(
+            repeater_id=self.repeater_id,
+            repeater_type=self.repeater_type,
+            domain=self.domain,
+            registered_on=now,
+            next_check=now,
+            payload_id=payload.get_id
+        )
+        metrics_counter('commcare.repeaters.new_record', tags={
+            'domain': self.domain,
+            'doc_type': self.doc_type,
+            'mode': 'sync' if fire_synchronously else 'async'
+        })
+        repeat_record.save()
+
+        if fire_synchronously:
+            # Prime the cache to prevent unnecessary lookup. Only do this for synchronous repeaters
+            # to prevent serializing the repeater in the celery task payload
+            RepeatRecord.repeater.fget.get_cache(repeat_record)[()] = self
+
+        repeat_record.attempt_forward_now(fire_synchronously=fire_synchronously)
+        return repeat_record
+
+    def allowed_to_forward(self, payload):
+        """
+        Return True/False depending on whether the payload meets forawrding criteria or not
+        """
+        return True
+
+    def pause(self):
+        self.paused = True
+        self.save()
+
+    def resume(self):
+        self.paused = False
+        self.save()
+
+    def retire(self, sync_to_couch=True):
+        self.paused = False
```

Comment:
Just curious, do you know why the repeater is unpaused when it is retired?

Reply:
I found this PR https://github.com/dimagi/commcare-hq/pull/22767 which says

> The repeater in this domain had been paused before being deleted, so deleting the records was just being postponed by an hour every time the trigger fired.

Reply:
Out of scope for this PR, but I wonder if the repeater logic should be updated to discard records for repeaters that have been deleted rather than postponing them?

Reply:
I think that can be handelled when we start migrating RepeatRecords.